parser/openfoodfacts_taxonomy_parser/parser/taxonomy_parser.py

```diff
@@ -162,12 +165,12 @@ def _header_harvest(self, filename: str) -> tuple[list[str], int]:
 
     def _entry_end(self, line: str, data: NodeData) -> bool:
         """Return True if the block ended"""
-        # stopwords and synonyms are one-liner, entries are separated by a blank line
-        if line.startswith("stopwords") or line.startswith("synonyms") or not line:
-            # can be the end of an block or just additional line separator,
-            # file_iter() always end with ''
-            if data.id:  # to be sure that it's an end
-                return True
+        if data.id.startswith("stopwords") or data.id.startswith("synonyms"):
+            # stopwords and synonyms are one-liners; if the id is set, it's the end
```

Comment:
It's possible, but most of the time synonyms and stopwords are written one after the other with no blank lines between them (because they are one-liners so they don't need to be separated with blank lines), so you can't really know which line to assign comments to.
I think it's better to determine that comments should be written before and keep only the comments above, as this structures the file and makes it possible to rewrite the taxonomy identically on export, which is the aim here.

Reply:
Ok it makes sense thanks!